src/Strategy/Composer.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Strategy.Composer
+  ( discover,
+    analyze,
+    buildGraph,
+    ComposerLock (..),
+    CompDep (..),
+    Source (..),
+  )
+where
+
+import Control.Effect.Diagnostics
+import qualified Data.Map.Strict as M
+import DepTypes
+import Discovery.Walk
+import Effect.Grapher
+import Effect.ReadFS
+import Graphing (Graphing)
+import Prologue
+import Types
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \_ _ files -> do
+  case find (\f -> fileName f == "composer.lock") files of
+    Nothing -> pure ()
+    Just file -> runSimpleStrategy "php-composerlock" PHPGroup $ analyze file
+
+  pure WalkContinue
+
+data ComposerLock = ComposerLock
+  { packages :: [CompDep],
+    packagesDev :: [CompDep]
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+data CompDep = CompDep
+  { name :: Text,
+    version :: Text,
+    source :: Source,
+    -- | name to version spec
+    require :: Maybe (Map Text Text),
+    requireDev :: Maybe (Map Text Text)
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+data Source = Source
+  { sourceType :: Text,
+    url :: Text,
+    reference :: Text
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+instance FromJSON ComposerLock where
+  parseJSON = withObject "ComposerLock" $ \obj ->
+    ComposerLock <$> obj .: "packages"
+      <*> obj .: "packages-dev"
+
+instance FromJSON CompDep where
+  parseJSON = withObject "CompDep" $ \obj ->
+    CompDep <$> obj .: "name"
+      <*> obj .: "version"
+      <*> obj .: "source"
+      <*> obj .:? "require"
+      <*> obj .:? "require-dev"
+
+instance FromJSON Source where
+  parseJSON = withObject "Source" $ \obj ->
+    Source <$> obj .: "type"
+      <*> obj .: "url"
+      <*> obj .: "reference"
+
+analyze :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m ProjectClosureBody
+analyze file = mkProjectClosure file <$> readContentsJson @ComposerLock file
+
+mkProjectClosure :: Path Abs File -> ComposerLock -> ProjectClosureBody
+mkProjectClosure file lock =
+  ProjectClosureBody
+    { bodyModuleDir = parent file,
+      bodyDependencies = dependencies,
+      bodyLicenses = []
+    }
+  where
+    dependencies =
+      ProjectDependencies
+        { dependenciesGraph = buildGraph lock,
+          dependenciesOptimal = Optimal,
+          dependenciesComplete = Complete
+        }
+
+newtype CompPkg = CompPkg {pkgName :: Text}
+  deriving (Eq, Ord, Show, Generic)
```

Comment:
No need for this `Generic`, we've phased this out across the codebase in #124.
This will also require a `{-# LANGUAGE GeneralizedNewtypeDeriving #-}` declaration at the top of the module.

```suggestion
  deriving (Eq, Ord, Show)
```